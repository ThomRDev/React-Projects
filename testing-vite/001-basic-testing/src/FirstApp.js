import React from 'react'
import PropTypes from "prop-types"

export const FirstApp = ({ title,subtitle }) => {
  return (
    <>
      <h1 data-testid='test-title'>{ title }</h1>
      <h2>{ subtitle }</h2>
      <p>Mensaje</p>
      <p>Mensaje</p>
      <p>Mensaje</p>
    </>
  )
}

FirstApp.propTypes = {
  title:PropTypes.string.isRequired,
  subtitle:PropTypes.string,
}
FirstApp.defaultProps = {
  // title : "I'm a title",
  subtitle : "I'm a subtitle",
}