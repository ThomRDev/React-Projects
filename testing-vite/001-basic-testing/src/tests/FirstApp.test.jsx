import { render } from "@testing-library/react"
import { FirstApp } from "../FirstApp"
describe('Test in FirstApp component', () => {
  test('should do to match with the snapshot', () => {
    const { container } = render( <FirstApp title="Titulo" /> )
    expect(container).toMatchSnapshot()
  })
  
  test('should show the title in h1 tag', () => {
    const { container,getByText,getByTestId,getAllByText } = render( <FirstApp title="Titulo" /> )
    expect(getByText("Titulo")).toBeTruthy()

    expect(getByTestId('test-title')).toBeTruthy()
    expect(getByTestId('test-title').innerHTML).toBe("Titulo")
    // no es recomendado
    // const h1 = container.querySelector("h1")

    // expect(h1.innerHTML).toBe("Titulo")
    // expect(h1.innerHTML).toContain("Titulo")

    expect(getAllByText("Mensaje").length).toBe(3)

  })
})