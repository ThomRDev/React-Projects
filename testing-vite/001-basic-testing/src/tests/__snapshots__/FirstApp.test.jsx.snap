// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in FirstApp component should do to match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Titulo
  </h1>
  <h2>
    I'm a subtitle
  </h2>
  <p>
    Mensaje
  </p>
  <p>
    Mensaje
  </p>
  <p>
    Mensaje
  </p>
</div>
`;
